BlueprintUICommonControls/Sources/ScrollView.swift

```diff
@@ -485,3 +504,41 @@ extension ScrollerWrapperView : KeyboardObserverDelegate {
     }
 }
 
+
+extension ScrollView {
+
+    final public class ScrollTrigger {
+
+        var action: (ContentOffset, Bool) -> Void
+
+        public init() {
+            action = { (_, _) in }
+        }
+
+        public func scroll(to: ContentOffset, animated: Bool) {
+            action(to, animated)
+        }
+    }
+    
+    public struct ContentOffset {
+        
+        public struct ScrollingContext {
+            public var contentSize: CGSize
+            public var scrollViewBounds: CGRect
+            public var contentInsets: UIEdgeInsets
+        }
+
+        public typealias Provider = (ScrollingContext) -> CGPoint
+
+        public static let top = ContentOffset { _ in .zero }
+        public static let bottom = ContentOffset { context in
+            CGPoint(x: 0, y: context.contentSize.height - context.scrollViewBounds.size.height)
```

Comment:
Ah, I forgot to include insets my other comment — the y value here should probably include `+ context.contentInsets.bottom`